src/event.rs

```diff
@@ -629,10 +629,22 @@ pub struct KeyboardInput {
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub enum CompositionEvent {
-    CompositionStart(String),
-    CompositionUpdate(String, usize),
-    CompositionEnd(String),
+pub enum IMEEvent {
+    /// The user enables IME
+    Enabled,
+    /// Preedit session is going to begin with the value
+    PreeditStart(String),
+
+    /// The user updates preedit status on IME.
+    ///
+    /// The value represents a pair of the preedit string and the cursor position.
+    /// The cursor position is byte-wise indexed.
```

Comment:
We should document that the first argument is a cursor begin and second is cursor end. Also if your cursor being and end are equal you should send both of them the same, so in your code you should always send Some(position), Some(position)?

Also, both `None` should represent a hidden cursor, we should state it as well.

Reply:
Added those description